pywbem/cim_operations.py

```diff
@@ -1255,14 +1331,404 @@ class are to be included in the returned instances, as follows:
+                                   ContinueOnError=None, MaxObjectCount=None,
+                                   **extra):
+        # pylint: disable=invalid-name
+
+        """
+        TODO Fix this
+        """
+        if namespace is None and isinstance(ClassName, CIMClassName):
+            namespace = ClassName.namespace
+        namespace = self._iparam_namespace_from(namespace)
+        classname = self._iparam_classname(ClassName)
+
+        result = self._imethodcall(
+            'OpenEnumerateInstancePaths',
+            namespace,
+            ClassName=classname,
+            FilterQueryLanguage=FilterQueryLanguage,
+            FilterQuery=FilterQuery,
+            ContinueOnError=ContinueOnError,
+            MaxObjectCount=MaxObjectCount,
+            pull_op=True,
+            **extra)
+
+        # TODO ks 5/16 why are we setting the namespace attribute??
+        #[setattr(i.path, 'namespace', namespace) for i in instances]
+
+        # returns a tuple containing instances and enumeration state
+
+        rslt_tuple = self._process_pull_result(result)
+
+        enum_ctxt = EnumContext(rslt_tuple[1], rslt_tuple[2], namespace,
+                                MaxObjectCount)
+
+        rtn_tuple = (rslt_tuple[0], enum_ctxt)
+
+        return rtn_tuple
+
+
+
+    def PullInstancesWithPath(self, enum_context, MaxObjectCount=None,
+                              **extra):
+        # pylint: disable=invalid-name
+
+        """
+        Retrieve the next set of instances from an open enumeratonsession
+        defined by the `enumeration_context` parameter.  The retrieved
+        instances include their instance paths.
+
+        This method performs the PullInstanceWithPath operation
+        (see :term:`DSP0200`).
+
+        If the operation succeeds, this method returns.
+        Otherwise, this method raises an exception.
+
+        :Parameters:
+          enum_context (TODO)
+            The sequence state variable returned from the previous
+            operation (open or pull) for the enumeration sequence.
+
+          MaxObjectCount (term: `Uint32`)
+            Defines a value that sets the maximum number of instances the
+            WBEM server can return for this request. any Uint32 number
+            including zero is valid. This is used only to change the value
+            from a previous request of the sequence.  If the MaxObjectCount
+            is to be the same as the previous request it is already defined
+            bu the enum_sequence_state variable.
+            * If non-zero `Uint32` the WBEM server must return no more than the
+              number of instances defined by this value.
+            * If zero the WBEM server must return zero instances.  This may
+              be used by a client to restart the timeout.
+            * If `None` the method uses the max_object_count contained in
+              the enum_sequence_state object.
```

Comment:
Here we see one of the disadvantages of having a default max countvin the context: None is now used to indicate the enum context default and can no longer bexused to indicate using the server defined default. one important use case for pywbem is server testing.